File: augur-engine/src/collect/yahoo.ts
import type { Signal } from '../config/types.js';

const YAHOO_QUOTE_API = 'https://query1.finance.yahoo.com/v7/finance/quote';

const MARKET_TICKERS = [
  '^GSPC',    // S&P 500
  '^DJI',     // Dow Jones
  '^IXIC',    // NASDAQ
  '^STOXX50E',// Euro Stoxx 50
  '^N225',    // Nikkei 225
  'GC=F',     // Gold futures
  'CL=F',     // Crude oil futures
  'DX-Y.NYB', // USD index
  '^VIX',     // VIX
  '^TNX',     // 10Y Treasury yield
];

interface QuoteResult {
  symbol: string;
  shortName?: string;
  regularMarketPrice?: number;
  regularMarketChange?: number;
  regularMarketChangePercent?: number;
  regularMarketTime?: number;
}

/** Collect market signals from Yahoo Finance */
export async function collectYahoo(): Promise<Signal> {
  const symbols = MARKET_TICKERS.join(',');
  const res = await fetch(`${YAHOO_QUOTE_API}?symbols=${encodeURIComponent(symbols)}`, {
    headers: {
      'User-Agent': 'augur-engine/0.1',
    },
  });

  if (!res.ok) {
    throw new Error(`Yahoo Finance API error: ${res.status} ${res.statusText}`);
  }

  const data = (await res.json()) as { quoteResponse?: { result?: QuoteResult[] } };
  const quotes = (data.quoteResponse?.result ?? []).map((q) => ({
    symbol: q.symbol,
    name: q.shortName ?? q.symbol,
    price: q.regularMarketPrice,
    change: q.regularMarketChange,
    changePct: q.regularMarketChangePercent,
  }));

  return {
    source: 'yahoo',
    fetchedAt: new Date().toISOString(),
    content: quotes,
  };
}

File: augur-engine/src/collect/yahoo.py
"""Yahoo Finance market data collector."""

from __future__ import annotations

from datetime import datetime, timezone

import httpx

from ..config.types import Signal

YAHOO_QUOTE_API = "https://query1.finance.yahoo.com/v7/finance/quote"
MARKET_TICKERS = [
    "^GSPC", "^DJI", "^IXIC", "^STOXX50E", "^N225",
    "GC=F", "CL=F", "DX-Y.NYB", "^VIX", "^TNX",
]


async def collect_yahoo() -> Signal:
    symbols = ",".join(MARKET_TICKERS)

    async with httpx.AsyncClient(timeout=15) as client:
        resp = await client.get(
            YAHOO_QUOTE_API,
            params={"symbols": symbols},
            headers={"User-Agent": "augur-engine/0.1"},
        )
        resp.raise_for_status()
        data = resp.json()

    quotes = [{
        "symbol": q.get("symbol"),
        "name": q.get("shortName", q.get("symbol")),
        "price": q.get("regularMarketPrice"),
        "change": q.get("regularMarketChange"),
        "change_pct": q.get("regularMarketChangePercent"),
    } for q in data.get("quoteResponse", {}).get("result", [])]

    return Signal(
        source="yahoo",
        fetched_at=datetime.now(timezone.utc).isoformat(),
        content=quotes,
    )
